Add ACC 1.2.13 support with network configuration, processor management, and standalone installation [04.02.2026] (#339)
IBM/z_ansible_collections_samples: z_appliance_control_center/acc_install_ansible/README.md

# Appliance Control Center Installer Ansible Script

![ACC Installation](../images/acc_install.png)

In essence, there are 3 ways to install ACC:

1. Use the HMC's UI to install ACC. This way is useful if you have familiarity
   with the HMC's UI.
2. Use the provided ansible playbook `00_acc_install.yaml` to install ACC
   by sending commands from the control node (the machine that will run the
   `00_acc_install.yaml` playbook) to the HMC. This way is helpful in storing

>    the infrastructure information in form of variables and automated install of ACC.

Reviewer: ```suggestion
   `00_acc_install.yaml` playbook) to the HMC. This way is helpful in storing
   the infrastructure information in the form of variables and automating the install of ACC.
```

To install ACC, the following actions must be performed on the HMC
and your control node.

- Download the ACC installation image from Fix Central and store it on your
  control node (i.e., your laptop).
- Download this directory (`acc_install_ansible`) on
  the control node (e.g., a laptop), which will connect with ACC and if required, with
  the HMC.
- Modify the `acc_env_vars.yaml` file.
- If the control node can communicate with the HMC, then set the variable `hmc_connected`
  to `true` in the `acc_env_vars.yaml` file. Otherwise, keep it to `false`.
- With `hmc_connected: true`, ensure that control node (your laptop) and ACC IP address have sufficient
  authority and are added to the appropriate HMC whitelist to allow for proper
  communication to the HMC.
- With `hmc_connected: false`, login to HMC and perform appropriate actions like:
  - Ensure that in the HMC, the Secure Service Container (SSC) based ACC LPAR
    activation profile is created and is updated with correct values of network
    settings. (`chpid`, `prefix`, `fid`, etc.) and storage (initial 16 GB storage is required for ACC).
  - For DPM based machine, also connect the storage group to the partition.

>   - Activate the LPAR in SSC installer mode.

Reviewer: ```suggestion
- When `hmc_connected: false` has been set, be sure to login to HMC and perform the appropriate actions:
  - Ensure that in the HMC, the Secure Service Container (SSC) based ACC LPAR
    activation profile is created and updated with the correct network settings (`chpid`, `prefix`, `fid`, etc.) and
    has been given sufficient storage (initial 16 GB storage is required for ACC).
  - For DPM based machines, be sure to connect the storage group to the partition.
  - Activate the LPAR in SSC installer mode.
```

>   - Provide the IP of the ACC LPAR to the `LPAR_IP` variable in the `acc_env_vars.yaml` file.

Reviewer: Don't you always have to do this, regardless if `hmc_connected` is set to `true` or `false`? Maybe don't mention this here because it almost seems like this is something you need to do only when `hmc_connected: false`

Author: With the new release, it is not needed to activate the LPAR. Only if `hmc_connected: false`, we need the LPAR to be activate in SSC Installer mode.

Author: Ah, I get your point. In the DPM mode machines, we always need to attach the storage group. I will fix this.

  export ACC_ADMIN_DEFAULT_PASSWORD=<admin_password>
  ```

  The `admin_username` is actually the ACC's `lpar_username` and
  the `admin_password` is the ACC's `lpar_password`. These are the
  ACC's SSC LPAR's credentials. With `hmc_connected: true`, the playbook
  will automatically set the credentials of the LPAR and install ACC on the
  LPAR. With `hmc_connected: false`, these credentials must be provided by
  the HMC admin who has activated the LPAR in SSC Installer mode. These
  credentials are equal to `Administrator user ID` and `Administrator password`

>   on the image activation profile on the HMC.

Reviewer: New explicit verbiage:
```suggestion
  LPAR. With `hmc_connected: false`, do not export the above `HMC_*` and `ACC_*` credentials into your environment. These credentials must be provided by the HMC admin who has activated the LPAR in SSC Installer mode and represents the `Administrator user ID` and `Administrator password` configured in the image activation profile on the HMC.
```

Author: @frankmit11 we still have to export the `ACC_ADMIN_USER` and `ACC_ADMIN_DEFAULT_PASSWORD` credentials even with `hmc_connected: false` as these values are used to access the LPAR using LPAR's IP.

      FCP variables are properly configured. Set the `IS_FCP` variable to true, and specify
      the `lun` and `wwpn` values in `the acc_env_vars.yaml` file.
      - **Note**: When using FCP, the `DISK_ID` variable represents the FCP device number used to communicate with the `lun` on the storage controller.
  - Change the `NW_IP`, `NW_FID`, `NW_GW_IP` and other network related
    variables in the `acc_env_vars.yaml` file.
    - If you plan to use FID instead of a OSA card, then do not keep
      `NW_FID` empty. If `NW_FID` variable is empty, then the scripts

>       will use the `NW_CHPID`, `NW_PORT` for the ACC LPAR.

Reviewer: Is this only required for DPM mode enabled machines?

Author: No, this is also required for non-DPM (PR/SM) mode machines as well.
